site/app/controllers/PollController.php

```diff
@@ -538,6 +547,13 @@ public function viewResults($poll_id) {
             );
         }
         $poll = $this->core->getQueries()->getPoll($poll_id);
+
+        if (!$this->core->getUser()->accessAdmin() && !$poll->isStudentHistogramAvailable()) {
+            return MultiResponse::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['polls']))
+            );
+        }
```

Comment:
I tested trying to view the poll histogram page from a student after deleting the poll as an instructor and got the error: 
![image](https://user-images.githubusercontent.com/71195502/132605612-0c976e30-84f7-4e95-be42-cd4dabf804d5.png)
I believe you also want to check if `$poll` is not null here, since this is what the function `getPoll` returns when no row in the DB matches the given id. This way, this if statement will also cover this case and students won't see a fog robot page, but would instead be redirected to the main poll page.

Reply:
Fixed.